### DISCUSSION TITLE
Authentication for clients to post events.

### DISCUSSION THREAD

**Discussion by toshanmugaraj** [plausible/analytics#3331] [Self-Hosted Support]
Can I configure secret API key so that clients are authenticated to post events. Now I see any one can post events if they have the server url

**Comment by ruslandoga** [comment]
👋 @toshanmugaraj 

I don't think that is supported right now. But even then, it would be difficult to make that secret key secret since it'd still be visible on your websites.

**Comment by toshanmugaraj** [comment]
Its from a mobile app, so the secret key will not be visible. I decided to validate that in the nginx proxy. But it would be great to have an option inbuilt within plausible.

**Reply by ruslandoga** [reply]
It's still probably visible :) https://nshipster.com/secrets/#normal-brain-hard-code-secrets-in-source-code

**Reply by ruslandoga** [reply]
Great that you've been able to solve it with nginx! Would you be able to share your config here?

**Reply by Daniel15** [reply]
> It's still probably visible

@ruslandoga Some basic obfuscation is usually sufficient enough, since it means people have to attach a debugger to find the secret key, which is a lot more effort than just running `strings` on the executable.

**Comment by baparham** [comment]
This is an interesting question though, is a normal plausible.io site vulnerable to something like a DDOS attack just flooding your stats with bogus data? I suppose the value there is low for an attacker, therefore the risk of such a vulnerability is low, but I've been pondering how it's possible to just pick any random domain to send stats too, even if that's not where the tracker is actually being run from.

Is proxying events through a separate server how most people at mitigating this?

**Reply by Daniel15** [reply]
It's tricky. By definition, if the logging is client-side, it's the client that decides if an event needs to be logged. Someone wanting to fill the logs with junk can just call the log function in the browser dev tools (or hit the API directly, of course).

For large-ish floods, Plausible may have some basic rate limiting built-in; I'm unsure. Nginx can rate-limit requests using the `limit_req_zone` and `limit_req` config directives, and other web servers can do similar things, but the "right" limits would depend on how many events you receive per second overall as well as how many events you expect from a 'regular' user per second. Nginx is pretty flexible - You choose the key to rate limit by and can have multiple rate limits configured for a single `server` or `location` block, so for example you can have both a per-IP rate limit as well as a per-endpoint or per-hostname rate limit.

For protection against large DDoS attacks (not just for Plausible but for any web service), you probably want dedicated protection (e.g. Path.net, Voxility, Cloudflare Magic Transit, F5 BIG-IP, etc)
